bin/semantic-release.js

```diff
@@ -91,59 +91,55 @@ npmconf.load({}, function (err, conf) {
         if (!options.debug) process.exit(1)
       }
 
-      var nerfDart = require('nerf-dart')(npm.registry)
+      var nerfDart = require('nerf-dart')(npmConf.registry)
       var wroteNpmRc = false
 
       if (env.NPM_OLD_TOKEN && env.NPM_EMAIL) {
         // Using the old auth token format is not considered part of the public API
         // This might go away anytime (i.e. once we have a better testing strategy)
-        conf.set('_auth', '${NPM_OLD_TOKEN}', 'project')
-        conf.set('email', '${NPM_EMAIL}', 'project')
+        npm.set('_auth', '${NPM_OLD_TOKEN}', 'project')
```

Comment:
Another fun fact: `npm.config.set`, does not write the config value to disk. It only writes it into memory within the context of the `npm.config` instance. However, `npm.set` will write the value to the user's `.npmrc` file.